README.md

```diff
@@ -1,51 +1,187 @@
+replication status attached (so equivalent to
+`TARGET_REPLICATION_STATUS=NEW`).
+
+Examples:
+
+`TARGET_REPLICATION_STATUS=PENDING,COMPLETED`
+
+Requeue objects that either have a replication status of PENDING or
+COMPLETED for CRR, do not requeue the others.
+
+`TARGET_REPLICATION_STATUS=NEW,PENDING,COMPLETED,FAILED`
+
+Trigger CRR on all original source objects (not replicas) in a bucket.
+
+`TARGET_REPLICATION_STATUS=REPLICA`
+
+For disaster recovery notably, it may be useful to reprocess REPLICA 
+objects to re-sync a backup bucket to the primary site.
+
+#### WORKERS
+
+Specify how many parallel workers should run to update object
+metadata. The default is 10 parallel workers.
+
+#### MAX_UPDATES
+
+Specify a maximum number of metadata updates to execute before
+stopping the script.
+
+If the script reaches this limit, it outputs a log line containing
+the KeyMarker and VersionIdMarker to pass to the next invocation (as
+environment variables `KEY_MARKER` and `VERSION_ID_MARKER`) and the
+updated bucket list without the already completed buckets. At the next
+invocation of the script, those two environment variables must be
+set and the updated bucket list passed on the command line to resume
+where the script stopped.
+
+The default is unlimited (will process the complete listing of buckets
+passed on the command line).
+
+**If the script queues too many objects and Backbeat cannot
+ process them quickly enough, Kafka may drop the oldest entries**,
+ and the associated objects will stay in the **PENDING** state
+ permanently without being replicated. When the number of objects
+ is large, it is a good idea to limit the batch size and wait 
+ for CRR to complete between invocations.
+
+Example:
+
+`MAX_UPDATES=10000`
+
+This limits the number of updates to 10,000 objects, which requeues
+a maximum of 10,000 objects to replicate before the script stops.
+
+#### KEY_MARKER
+
+Set to resume from where an earlier invocation stopped (see
+[MAX_UPDATES](#MAX_UPDATES)).
+
+#### VERSION_ID_MARKER
+
+Set to resume from where an earlier invocation stopped (see
+[MAX_UPDATES](#MAX_UPDATES)).
+
+### Example use cases
+
+#### CRR existing objects after setting a replication policy for the first time
+
+For this use case, it's not necessary to pass any extra environment
+variable, because the default behavior is to process objects without a
+replication status attached.
+
+To avoid requeuing too many entries at once, pass this value:
+
+```
+export MAX_UPDATES=10000
+```
+
+#### Re-queue objects stuck in PENDING state
+
+If Kafka has dropped replication entries, leaving objects stuck in a
+PENDING state without being replicated, pass the following extra
+environment variables to reprocess them:
+
+```
+export TARGET_REPLICATION_STATUS=PENDING
+export MAX_UPDATES=10000
+```
+
+**Warning**: This may cause replication of objects already in the 
+Kafka queue to repeat. To avoid this, purge the "backbeat-replication"
```

Comment:
I think instead of purging, we can mention setting the offsets to skip over the consumer log.